# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_coreutils.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

I guess non-pointer FILE is not portable at least it breaks with dietlibc ...

  - Rene Rebe <rene@exactcode.de>

--- coreutils-5.2.1/src/paste.c.vanilla	2005-06-15 23:48:26.000000000 +0200
+++ coreutils-5.2.1/src/paste.c	2005-06-15 23:48:49.000000000 +0200
@@ -52,11 +52,11 @@
 /* Indicates that no delimiter should be added in the current position. */
 #define EMPTY_DELIM '\0'
 
-static FILE dummy_closed;
+static FILE* dummy_closed;
 /* Element marking a file that has reached EOF and been closed. */
-#define CLOSED (&dummy_closed)
+#define CLOSED (dummy_closed)
 
-static FILE dummy_endlist;
+static FILE* dummy_endlist;
 /* Element marking end of list of open files. */
 #define ENDLIST (&dummy_endlist)
 
--- coreutils-5.2.1/lib/closeout.c	2003-07-19 10:04:28.000000000 +0200
+++ coreutils-5.2.1-fixed/lib/closeout.c	2006-01-04 17:49:42.000000000 +0100
@@ -77,7 +77,7 @@
 
   /* If the stream's error bit is clear and there is nothing to flush,
      then return right away.  */
-  if (e && __fpending (stdout) == 0)
+  if (e /* && __fpending (stdout) == 0 */)
     return;
 
   if (fclose (stdout) != 0)
